sdk/core/azure_core/src/error/error_response.rs

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore innererror
+
+use crate::{
+    error::{Error, ErrorKind},
+    http::{headers::ERROR_CODE, BufResponse},
+};
+use serde::Deserialize;
+use std::{collections::HashMap, str};
+
+/// An HTTP error response.
+///
+/// Implements a standard "ErrorResponse" as described in the [API guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#handling-errors).
+///
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ErrorResponse {
+    error: Option<ErrorDetail>,
+}
+
+impl ErrorResponse {
+    /// The error details.
+    pub fn error(&self) -> Option<&ErrorDetail> {
+        self.error.as_ref()
+    }
+}
+
+impl TryFrom<Error> for ErrorResponse {
```

Comment:
Note that Cosmos's tests depend on having the ability to convert an `Error` to a `ErrorResponse`, and because the `RawResponse` is wrapped in an `Option`, we can only borrow, which means that calling `.json()` on the `body()` fails because `.json()` consumes self.

The `try_from()` call neatly avoids that by directly calling `serde_json` like the other logic here does, thus encapsulating the serde logic from the rest of the SDK.